omnibus/files/private-chef-cookbooks/private-chef/libraries/private_chef.rb

```diff
@@ -495,8 +603,48 @@ def gen_ldap
 
     end
 
+
+    # True if the given topology requires per-server config via `server` blocks
+    def server_config_required?
+      PrivateChef["topology"] == "ha" ||
+        PrivateChef["topology"] == "tier" ||
+        (PrivateChef["registered_extensions"][PrivateChef["topology"]] &&
+         PrivateChef["registered_extensions"][PrivateChef["topology"]][:server_config_required])
+    end
+
+    def assert_server_config(node_name)
+      unless PrivateChef["servers"].key?(node_name)
+        Chef::Log.fatal <<-EOF
+No server configuration found for "#{node_name}" in /etc/opscode/chef-server.rb.
+Server configuration exists for the following hostnames:
+
+  #{PrivateChef["servers"].keys.sort.join("\n  ")}
+
+EOF
+        exit(1)
```

Comment:
exit! 1 to prevent a stack trace, at leas until we add our own hooks.  Maybe move this to a preflight check? (Or just note to do so in the future...)